sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1594,13 +1596,11 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map("path" -> path))
   }
 
-  def dataSourceOutputModeUnsupportedError(
-      className: String, outputMode: OutputMode): Throwable = {
+  def fileSinkOutputModeUnsupportedError(
```

Comment:
If it's not weird to have INVALID_STREAMING_OUTPUT_MODE.UNSUPPORTED_OPERATOR, this can be INVALID_STREAMING_OUTPUT_MODE.UNSUPPORTED_DATA_SOURCE. But again, this has to be flipped, like `DATA_SOURCE_UNSUPPORTED_STREAMING_OUTPUT_MODE`. (If there is a general error class group for data source, include there.)

Probably it might not be good to group them with INVALID_STREAMING_OUTPUT.

Reply:
Thank you for your very insightful suggestions, let me make some improvements.

Reply:
I have updated according to the review. Please take a look when you have time. @HeartSaVioR